### PR TITLE
Add VSTS CI for Windows

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

- Add VSTS CI for Windows
    - Disable `Access-denied test for Get-Item C:\windows\appcompat\Programs\Install -ErrorAction Stop`, because the path does not always exist
        - https://github.com/PowerShell/PowerShell/issues/7553
    - Disable `Should give .sys file if the fullpath is specified with hidden and force parameter`, because pagefile.sys doesn't always exist and other files don't meet test's requirement.
        - https://github.com/PowerShell/PowerShell/issues/7554
    - Disable some `Test-Connection` tests for same reasons they failed on VSTS Linux
        - https://github.com/PowerShell/PowerShell/issues/7555
    - Disable `Test-FileCatalog should pass when catalog is in the same folder as files being tested`, because the CmdLet does not work in that scenario
         - Also, give details needed to investigate when test fails
         - https://github.com/PowerShell/PowerShell/issues/7556
    - Update `appveyor.psm1` to work with VSTS
    - Update `HelpersRemoting.psm1` `New-RemoteSession` to work for CimSession (discovered an issue during the investigation)
    - Update `Test wildcard with drive relative directory path` to work when there are multiple drives
         - Disable on non-windows machines since the test is assuming drive letters
    - Update `New-CimSession` Tests to requireAdmin
         - Also, make sure session name is a string
    - Add functions to save and restore psoptions
    - update `.gitatttributes` so files clone like they do on appveyor

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-ChildItem.Tests.ps1 [hunk 1]
```diff
@@ -161,7 +161,8 @@ Describe "Get-ChildItem" -Tags "CI" {
             (Get-ChildItem -Path $searchRoot -Directory -Recurse).Count | Should -Be 1
         }
 
-        It "Should give .sys file if the fullpath is specified with hidden and force parameter" -Skip:(!$IsWindows) {
+        # VSTS machines don't have a page file
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Get-ChildItem.Tests.ps1]: at all or on system drive?

Comment (TravisEz13) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Get-ChildItem.Tests.ps1]: They don't have a page file on the system drive

File: test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1 [hunk 2]
```diff
@@ -79,7 +79,7 @@ Describe "Test-Connection" -tags "CI" {
         }
 
         # In VSTS, address is 0.0.0.0
-        It "Force IPv4 with implicit PingOptions" -Skip:(Test-IsVstsLinux) {
+        It "Force IPv4 with implicit PingOptions" -Skip:((Test-IsVstsLinux) -or (Test-IsVstsWindows)) {
```
[the remark this answers is not in the document]
Comment (TravisEz13) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1]: remoting tests are working

[the remark this answers is not in the document]
Comment (TravisEz13) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1]: At this point, only CI has been ported, not Feature tests.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-ChildItem.Tests.ps1 (hunk 1 above)
Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Get-ChildItem.Tests.ps1]: We could find the page file location:
```powershell
			#Open the registry on multiple remote computers
			$reg = [Microsoft.Win32.RegistryKey]::OpenRemoteBaseKey('LocalMachine',$server )
			$RegKeyPath= "SYSTEM\CurrentControlSet\Control\Session Manager\Memory Management"
			$pageFileKey=$reg.OpenSubKey($RegKeyPath)
			$pageFileLocation=$pageFileKey.GetValue("ExistingPageFiles")
```

File: build.psm1 [hunk 3]
```diff
@@ -2970,6 +2978,102 @@ function Restore-PSOptions {
...
+        $Remove
+    )
+
+    $options = Get-Content -Path $PSOptionsPath | ConvertFrom-Json
+
+    if($Remove)
+    {
+        # Remove PSOptions.
+        # The file is only used to set the PSOptions.
+        Remove-Item -Path $psOptionsPath
```
Comment (adityapatwardhan) [inline build.psm1]: Maybe `-force`

File: test/powershell/Modules/Microsoft.PowerShell.Security/FileCatalog.Tests.ps1 [hunk 4]
```diff
@@ -244,6 +249,12 @@ Describe "Test suite for NewFileCatalogAndTestFileCatalogCmdlets" -Tags "CI" {
                 Remove-Item "$env:temp\UserConfigProv\" -Force -ErrorAction SilentlyContinue -Recurse
             }
 
+            # We will fail, Write why.
+            if($detailResult)
+            {
+                $detailResult | ConvertTo-Json | Write-Verbose -Verbose
```
Comment (adityapatwardhan) [inline test/powershell/Modules/Microsoft.PowerShell.Security/FileCatalog.Tests.ps1]: This can be moved to L#242

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**